internal/providers/azure/pke/driver/commoncluster/cluster.go

```diff
@@ -305,3 +305,7 @@ func (a *AzurePkeCluster) GetCAHash() (string, error) {
 func (a *AzurePkeCluster) GetPKEOnAzureCluster() pke.Cluster {
 	return a.model
 }
+
+func (a *AzurePkeCluster) GetKubernetesVersion() (string, error) {
```

Comment:
I wonder if instead of getting the k8s version from the database could we query the version from Kubernetes itself as that is the source of truth.